src/features/quality-stats/QualityStatsCollector.js

```diff
@@ -111,13 +114,17 @@ class QualityStatsCollector {
      * the initial object structure will be created.
      * @param {Object} pcData - Object containing PC collected data, including multiple ssrc data
      * @param {string} ssrc - ssrc of the track from which we want to obtain collected data
+     * @param {number} timestamp - the UNIX timestamp of when this stats entry was observed on the client endpoint.
      * @returns {Object}`- Track collected data.
      */
-    _getTrackData(pcData, ssrc) {
-        if (!pcData[ssrc]) {
+    _getTrackData(pcData, ssrc, timestamp) {
+        if (pcData[ssrc]) {
+            // TODO what if endTime < startTime (client clock update)
+            pcData[ssrc].endTime = timestamp;
```

Comment:
I would protect this with a check if `timestamp` is not undefined. That should prevent setting `endTime` accidentally. And as a plus allows to call `_getTrackData()` without a timestamp if needed.

I would compare the start and end time and rather set them to the same values, rather then creating negative track durations by allowing `endTime < startTime`.